reference/api/settings.mdx

```diff
@@ -1910,3 +1911,333 @@ Reset an index's typo tolerance settings to their [default value](#typo-toleranc
+    "documentTemplate": "A movie titled {{doc.title}} whose description starts with {{doc.overview|truncatewords: 20}}",
+    "dimensions": 1536
+  }
+}
+```
+
+### Update embedder settings
+
+<RouteHighlighter method="PATCH" route="/indexes/{index_uid}/settings/embedders"/>
+
+Partially update the embedder settings for an index. When this setting is updated Meilisearch may reindex all documents and regenerate their embeddings.
+
+#### Path parameters
+
+| Name              | Type   | Description                                                               |
+| :---------------- | :----- | :------------------------------------------------------------------------ |
+| **`index_uid`** * | String | [`uid`](/learn/core_concepts/indexes#index-uid) of the requested index |
+
+#### Body
+
+```json
+{
+  "default": {
+    "source": <String>,
+    "url": <String>,
+    "apiKey": <String>,
+    "model": <String>,
+    "documentTemplate": <String>,
+    "dimensions": <Integer>, 
+    "revision": <String>,
+    "inputField": [<String>, <String>, …],
+    "inputType": "text"|"textArray", 
+    "query": <Object>,
+    "pathToEmbeddings": [<String>, <String>, …],
+    "embeddingObject": [<String>, <String>, …], 
+    "distribution": {
+      "mean": <Float>,
+      "sigma": <Float>
+    }
+  }
+}
+```
+
+Set an embedder to `null` to remove it from the embedders list.
+
+##### `source`
+
+Use `source` to configure an embedder's source. The following embedders can auto-generate vectors for documents and queries:
+
+- `openAi`
+- `huggingFace`
+- `ollama`
+
+Additionally, use `rest` to auto-generate embeddings with any embedder offering a REST API.
+
+You may also configure a `userProvided` embedder. In this case, you must manually include vector data in your documents' `_vector` field. You must also manually generate vectors for search queries.
+
+This field is mandatory.
+
+##### `url`
+
+Meilisearch queries `url` to generate vector embeddings for queries and documents. `url` must point to a REST-compatible embedder.
+
+This field is mandatory when using `rest` embedders.
+
+This field is optional when using `ollama` embedders.
+
+This field is incompatible with `openAi`, `huggingFace`, and `userProvided` embedders.
+
+##### `apiKey`
+
+Authentication token Meilisearch should send with each request to the embedder.
+
+This field is mandatory if using a protected `rest` embedder.
+
+This field is optional for `openAI` and `ollama` embedders. If you don't specify `apiKey`, Meilisearch will attempt to read it from environment variables `OPENAI_API_KEY` and `MEILI_OLLAMA_URL`, respectively.
+
+This field is incompatible with `huggingFace` and `userProvided` embedders.
+
+##### `model`
+
+The model your embedder uses when generating vectors. These are the officially supported models Meilisearch supports:
+
+- `openAi`: `openai-text-embedding-ada-002`, `text-embedding-3-small`, and `text-embedding-3-large`
+- `huggingFace`: `BAAI/bge-base-en-v1.5`
```

Comment:
For huggingFace, we should maybe explicitly indicate that other BERT models may work.

Reply:
I want to keep the list restricted to stuff we know and have tested. 

Since that doesn't seem to be the case for all BERT models, I'd rather add it to the paragraph below as a "it may work, but use it at your own peril".